src/main/java/org/tomitribe/auth/signatures/Signature.java

```diff
@@ -225,33 +354,118 @@ public AlgorithmParameterSpec getParameterSpec() {
+             */
+            private Object value;
+            /**
+             * A flag indicating whether the field is a string or number.
+             */
+            private boolean isNumber;
+
+            FieldValue(String value, boolean isNumber) throws ParseException {
+                this.isNumber = isNumber;
+                if (isNumber) {
+                    this.value = NumberFormat.getInstance().parse(value);
+                } else {
+                    this.value = value;
+                }
+            }
+
+            /** Returns true if the field is a string */
+            boolean isString() { return !isNumber; }
+            /** Returns true if the field is a number */
+            boolean isNumber() { return isNumber; }
+            /** Returns true if the field is an integer value */
+            boolean isInteger() { return value instanceof Long; }
+
+            /** Returns the field as a string, or null if the field is not a string. */
+            String getValueAsString() {
+                if (!isString()) return null;
+                return (String)value;
+            }
+
+            /** Returns the field as a long value, or null if the field is not a integer number. */
+            Long getValueAsLong() {
+                if (!isInteger()) return null;
+                return ((Number)value).longValue();
+            }
+            /** Returns the field as a double value, or null if the field is not a number. */
+            Double getValueAsDouble() {
+                if (!isNumber()) return null;
+                return ((Number)value).doubleValue();
+            }
+
+        }
+
         try {
             authorization = normalize(authorization);
 
-            final Map<String, String> map = new HashMap<String, String>();
+            final Map<String, FieldValue> map = new HashMap<String, FieldValue>();
 
             final Matcher matcher = RFC_2617_PARAM.matcher(authorization);
             while (matcher.find()) {
-                final String key = matcher.group(1).toLowerCase();
-                final String value = matcher.group(2);
-                map.put(key, value);
+                final String key = matcher.group("key").toLowerCase();
+                // The field value may be a double-quoted string or a number.
+                boolean isNumber = false;
+                String value = matcher.group("stringValue");
+                if (value == null) {
+                    value = matcher.group("numberValue");
+                    isNumber = true;
+                }
+                map.put(key, new FieldValue(value, isNumber));
             }
 
             final List<String> headers = new ArrayList<String>();
-            final String headerString = map.get("headers");
-            if (headerString != null) {
-                Collections.addAll(headers, headerString.toLowerCase().split(" +"));
+            FieldValue fv = map.get("headers");
```

Comment:
Minor comment. No need to change.
I would have used a meaningful name and let the IDE do the completion for you.

fieldValue instead of fv is more readable.

Reply:
sure will do